### PR TITLE
Fix: ensure that node is displayed if recorded

### PR DESCRIPTION
**Pull request by hmijail**
Signed-off-by: Horacio Mijail Anton Quiles <hmijail@gmail.com>

## Fixed Issue(s)
When acceptance tests used the ThreadBesuNodeRunner, the node information was not being logged

### CODE REVIEW

File: acceptance-tests/tests/src/test/resources/log4j2.xml [hunk 1]
```diff
@@ -5,7 +5,10 @@
   </Properties>
   <Appenders>
     <Console name="Console" target="SYSTEM_OUT">
-      <PatternLayout pattern="%X{test} | %d{HH:mm:ss.SSS} | %t | %-5level | %c{1} | %msg%n" />
+      <PatternLayout>
+        <pattern>%X{test} | %X{node} | %d{HH:mm:ss.SSS} | %t | %-5level | %c{1} | %msg%n</pattern>
+        <replace regex="\|  \| " replacement="| "/>
```
Comment (CjHare) [inline acceptance-tests/tests/src/test/resources/log4j2.xml]: What the side affect of this change, need a regex applied on log line strikes me as unusual

Comment (hmijail) [reply acceptance-tests/tests/src/test/resources/log4j2.xml]: The regex replacement is just to have cleaner log lines.
If no node information exists at the time of the logging, 2 separators `|` would appear together, like in the first line of these 2:
```
shouldNotDisplayDisabledServices |  | 21:38:17.681 | Test worker | INFO  | Cluster | Starting node nodeB (id = 372a...bafb)
shouldNotDisplayDisabledServices | nodeB | 21:38:17.685 | Test worker | INFO  | RocksDBKeyValueStorageFactory | No existing database detected at /var/folders/fg/txwyt5v92fncr6n02trmjwv40000gs/T/acctest2010709804941575271. Using version 1
```

The replacement removes the duplicate separator in that case:
```
shouldNotDisplayDisabledServices | 21:38:17.681 | Test worker | INFO  | Cluster | Starting node nodeB (id = 372a...bafb)
shouldNotDisplayDisabledServices | nodeB | 21:38:17.685 | Test worker | INFO  | RocksDBKeyValueStorageFactory | No existing database detected at /var/folders/fg/txwyt5v92fncr6n02trmjwv40000gs/T/acctest2010709804941575271. Using version 1
```

Personally I even prefer having the explicit empty field between the 2 separators, but I am trying to be extra-clean for upstream.

Comment (CjHare) [reply acceptance-tests/tests/src/test/resources/log4j2.xml]: That seems like an unfortunate design choice you're covering up by altering the output.

>If no node information exists at the time of the logging,

Why is there no node information with this logging? Perhaps different appender is a more appropriate choice, given different context/states?

Comment (hmijail) [reply acceptance-tests/tests/src/test/resources/log4j2.xml]: > That seems like an unfortunate design choice you're covering up by altering the output.

First of all I would note the empty/default/arbitrary initial content in other fields, like the thread name `%t`. IMO, having an empty field is clearer than having default filler.

But of course a trivial solution here is to add a default node name. "Null"?

> Why is there no node information with this logging? 

The acceptance test itself doesn't run in a node, so it has no node information. The node information is added to the ThreadContextMap the moment the ThreadBesuNodeRunner starts a new node. 

> Perhaps different appender is a more appropriate choice, given different context/states?

That was easy in the process runner scenario; I don't see how that could be done for the thread runner scenario.
AFAICT, the inheritable ThreadContextMap was purposefully made to do this kind of thing,

**Pull request merged**